roofit/roofitcore/inc/RooFuncWrapper.h

```diff
@@ -84,21 +88,38 @@ class RooFuncWrapper final : public RooAbsReal {
       }
       _gradientVarBuffer.reserve(_params.size());
 
+      // Extract observables
+      if (!ObsSet.empty()) {
+         auto dataSpans = data.getBatches(0, data.numEntries());
+
+         for (auto *obs : static_range_cast<RooRealVar *>(ObsSet)) {
+            RooSpan<const double> span{dataSpans.at(obs)};
```

Comment:
```suggestion
            RooSpan<const double> span{dataSpans.at(obs)};
            _observables.reserve(_observables.size() + data.numEntries());
```
To avoid frequent reallocation

Reply:
shouldn't it be '*' instead of '+'

Reply:
Maybe move this outside the loop with `_observables.reserve(_observables.size() * data.numEntries());` to truly save reallocation costs.

Reply:
Nice point :+1: On that note I have a question: the `RooFuncWrapper` only supports 1D data so far anyway right? So it doesn't make a difference so far where you would put the `reserve()`. Or does it works for a dataset with 2 columns?

Reply:
Ah, I am not sure what you mean by 'two columns' do you mean more than two observables having multiple entries?

Reply:
Yes, but it would be fine to support that later

Reply:
It is supported right now. As long as the string version of the function indexes its observable array properly, it should work fine.